examples/knative_postgresql_customvar/README.md

```diff
@@ -0,0 +1,252 @@
+cat <<EOS |kubectl apply -f -
+---
+kind: Namespace
+apiVersion: v1
+metadata:
+  name: service-binding-demo
+EOS
+```
+
+Alternatively, you can perform the same task with this make command:
+
+```shell
+make create-project
+```
+
+#### Import an application
+
+In this example we will import an arbitrary [Java Spring Boot application](https://github.com/ldimaggi/java-rest-http-crud).
+
+In the OpenShift Console switch to the Developer perspective. (Make sure you have selected the `service-binding-demo` project). Navigate to the `+ADD` page from the menu and then click on the `[Import from Git]` button. Fill in the form with the following:
+
+* `Git Repo URL` = `https://github.com/ldimaggi/java-rest-http-crud`
+* `Project` = `service-binding-demo`
+* `Application`->`Create New Application` = `java-app`
+* `Name` = `java-app`
+* `Builder Image` = `Java`
+* `Create a route to the application` = checked
+
+and click on the `[Create]` button.
+
+Notice, that during the import no DB config was mentioned or requestd.
+
+When the application is running navigate to its route to verify that it is up. Try the application's UI to add a fruit - it causes an error proving that the DB is not connected.
+
+#### Create a DB instance for the application
+
+Now we utilize the DB operator that the cluster admin has installed. To create a DB instance just create a `Database` custom resource in the `service-binding-demo` namespace called `db-demo`:
+
+```shell
+cat <<EOS |kubectl apply -f -
+---
+apiVersion: postgresql.baiju.dev/v1alpha1
+kind: Database
+metadata:
+  name: db-demo
+  namespace: service-binding-demo
+spec:
+  image: docker.io/postgres
+  imageName: postgres
+  dbName: db-demo
+EOS
+```
+
+Alternatively, you can perform the same task with this make command:
+
+```shell
+make create-backing-db-instance
+```
+
+#### Set labels on the application
+
+Now we need to set arbitrary labels on the application's `DeploymentConfig` in order for the Service Binding Operator to be able to find the application.
+
+The labels are:
+
+* `connects-to=postgres` - indicates that the application needs to connect to a PostgreSQL DB
+* `environment=demo` - indicates the demo environment - it narrows the search
+
+```shell
+kubectl patch dc java-app -p '{"metadata": {"labels": {"connects-to": "postgres", "environment":"demo"}}}'
+```
+
+Alternatively, you can perform the same task with this make command:
+
+```shell
+make set-labels-on-java-app
+```
+
+#### Express an intent to bind the DB and the application
+
+Now, the only thing that remains is to connect the DB and the application. We let the Service Binding Operator to 'magically' do the connection for us.
+
+Create the following `ServiceBindingRequest`:
+
+```shell
+cat <<EOS |kubectl apply -f -
+---
+apiVersion: apps.openshift.io/v1alpha1
+kind: ServiceBindingRequest
+metadata:
+  name: binding-request
+  namespace: service-binding-demo
+spec:
+  applicationSelector:
+    matchLabels:
+      connects-to: postgres
```

Comment:
Maybe let's use a resourceRef in this sample?

Reply:
@sbose78 Yes, good idea ;)

Reply:
@sbose78  This is tricky, assuming I want to use `resourceRef` instead of the `matchLabels`.

See: https://github.com/redhat-developer/service-binding-operator/issues/188